cleaner/environment_matrix/classes/clean.php

```diff
@@ -91,14 +96,47 @@ static public function execute() {
                         // set_config requires a null 'plugin' value when updating core configuration values.
                         $config->plugin = ($config->plugin == 'core') ? null : $config->plugin;
 
+                        // First, set config in database
                         if ($verbose) {
                             mtrace("set_config('{$config->config}', '{$config->value})'");
                         }
-
                         if (!$dryrun) {
                             set_config($config->config, $config->value, $config->plugin);
                         }
 
+                        // Generate an admin settings tree
+                        $admintree = admin_get_root(true);
+
+                        // Get strings in nicer format for reuse
+                        $configname = $config->config;
+                        $pluginname = $config->plugin;
+                        $elementname = $pluginname.$configname;
+
+                        // Search the admintree for configname
+                        $nodes = $admintree->search($configname);
+                        $relevantobject = '';
+
+                        // Iterate through tree for specific page and elementname
+                        foreach ($nodes as $node) {
+                            if ($node->page instanceof \admin_settingpage && isset($node->page->settings->$elementname)) {
+                                // Should only ever be reached once, so break loop
+                                $relevantobject = $node->page->settings->$elementname;
+                                break;
+                            }
+                        }
+
+                        // Now perform any additional validation
+                        if ($verbose) {
+                            // Show the additional write_settings
+                            mtrace("{$config->plugin}:{$relevantobject->name}->write_setting('{$config->value}')");
+                        }
+                        // Dont execute rewrite if dryrun
+                        if (!$dryrun && $relevantobject != '' && $relevantobject->plugin == $pluginname) {
+                            // Get setting object back out of config control
+                            $settings = $relevantobject->get_setting();
+                            // Reset to fire additional validation/actions
+                            $relevantobject->write_setting($settings);
```

Comment:
I think this can return errors? if so just log them and move on. I'd do this regardless of the dry run flag